Impedance type change, name change, bug fix

leewujung opened the pull request:
This PR contains the following changes:
- change the data type of impedance to `float` (addresses #1018)
- finish renaming `impedance_transmit` to `impedance_transducer` and `impedance_receive` to `impedance_transceiver` (leftover that resulted into bug from v0.7.0)
- rename transmit chirp replica construction methods: 
   - correct typo `Anderson` to `Andersen`
   - ~change `Macaulay` to `Matlab` since original code came from Andersen~
   - remove the previous `Macaulay` chirp replica implementation and only keep the one in reference to Andersen

emiliom commented:
> change `Macaulay` to `Matlab` since original code came from Andersen

I don't know anything about this, but an implementation name of "Matlab" seems not specific enough, unlike Andersen and Macaulay. The Andersen implementation code has a helpful 2-line comment that provides background. Is the "Matlab" implementation from a specific m-script file, from the general Matlab code base, or something else?

emiliom commented on echopype/calibrate/calibrate_ek.py:
```diff
@@ -407,6 +407,10 @@ def _get_power_from_complex(
             EchoData["Sonar/Beam_group1"] with selected channel subset
         chirp : dict
             a dictionary containing transmit chirp for BB channels
+        z_et : float
```
How about adding `float` type hints to `z_et` and `z_er`, while you're at it? Especially given that the other function arguments already have type hints.

leewujung replied:
Done!

leewujung commented:
@emiliom : I've added type hint to `z_et` and `z_er`. Also added a source reference to the gain correction based on angle offset and beamwidth that was undocumented anywhere else -- I asked Lars about it and added the explanation in the docstring.

Let me know what you think about the "Matlab" implementation of chirp.

emiliom commented:
> The "Matlab" implementation came from Lars Andersen in matlab Echolab code that was passed through multiple people (I got it from Chu).

:+1: 

> In fact, I think the best way to do this is to remove this implementation completely. I spoked with Lars about this and the specific chirp implementation is due to the implementation in hardware, so we should stick to it.

Well, the issue goes away if you remove the "Matlab" implementation :wink:. So, no additional comment here. It's up to you whether to keep it or not. If you do keep it, my suggestion would be to simply change the label to something like "MatlabEcholab" and add a comment like the sentence above describing its origin.

leewujung commented:
I think I'll opt for removing that section entirely, since it does not have the simplicity of just using a general chirp function, not does it conform with what Simrad actually uses. I'll push up a commit for the removal.
